client/components/data/query-atat-eligibility/index.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { flowRight as compose } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { requestEligibility } from 'state/automated-transfer/actions';
+
+export class QueryAutomatedTransferEligibility extends Component {
+	static propTypes = {
+		request: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+	};
+
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.request( nextProps );
+		}
+	}
+
+	request( { request, siteId } ) {
+		request( siteId );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export const mapDispatchToProps = dispatch => ( {
+	request: compose( dispatch, requestEligibility ),
+} );
+
+export default connect( null, mapDispatchToProps )( QueryAutomatedTransferEligibility );
```

Comment:
actually I was able to do it in this case just fine, though it did make testing more difficult because I had to manually pass the results into `dispatch()` but that's what `connect()` does I guess, so it's okay.

changed in 1b21165